core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -30,17 +30,23 @@
 import com.linecorp.armeria.common.HttpHeaders;
 import com.linecorp.armeria.common.HttpHeadersBuilder;
 import com.linecorp.armeria.common.RequestId;
+import com.linecorp.armeria.common.util.UnstableApi;
 
-class AbstractClientOptionsBuilder {
+/**
+ * A skeletal builder implementation for {@link ClientOptions}.
+ * Note: This class is meant for internal use only.
```

Comment:
I tried to move to the internal package first but `AbstractClientOptionsBuilder` access package-private method in `ClientDecoration`
https://github.com/line/armeria/blob/6178d35f4b26ae8fce31ede5d1b848aa3e2a80a6/core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java#L101-L102

I thought making this public with `@Unstable` annotation is better than change the method to `public`.
https://github.com/line/armeria/blob/8a67c5823ab3cfd1f4a27683409c638091093d3f/core/src/main/java/com/linecorp/armeria/client/ClientDecoration.java#L78-L84

Reply:
As a follow-up of this pull request, how about adding `ClientDecorationBuilder`, so that `AbstractClientOptionsBuilder` can mutate/build a `ClientDecoration` without accessing package-private methods?

Reply:
Then, we could move `AbstractrClientOptionsBuilder` to `internal`.

Reply:
> I tried to move to the internal package first but AbstractClientOptionsBuilder access package-private method in ClientDecoration

Ah, I didn't notice that. 😅

Reply:
> how about adding `ClientDecorationBuilder`, so that AbstractClientOptionsBuilder can mutate/build a ClientDecoration without accessing package-private methods?

IIUC, you were meant to add `addClientDecoration(ClientDecoration)` to [`ClientDecorationBuilder`](https://github.com/line/armeria/blob/8a67c5823ab3cfd1f4a27683409c638091093d3f/core/src/main/java/com/linecorp/armeria/client/ClientDecorationBuilder.java#L28)?

Sorry, I don't thoroughly understand your direction. 😅

Reply:
Oh, I completely forgot we have `ClientDecorationBuilder` already. :rofl: 
If it's enough with adding just `public ClientDecorationBuilder.add(ClientDecoration)`, then how about handling it in this PR?

Reply:
Sure, will handle it in this PR. :-)